src/data-access/facades/repository-facade.ts

```diff
@@ -250,22 +250,27 @@ export class RepositoryFacade {
     return new MongoSystemInformationRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()), new MongoEntityConverter(LoggerFacade.createLogger()))
   }
 
-  public static createDeviceDataChangedListener(): DataChangedListener<Device> {
+  public static createDeviceDataChangedListener(): DataChangedListener<CoreDevice> {
     return new MongoDeviceChangedListener(
       MongoDatabase.getInstance(LoggerFacade.createLogger()),
       new MongoEntityConverter(LoggerFacade.createLogger()),
       LoggerFacade.createLogger()
     )
   }
 
-  public static createDeviceRepository(): DeviceRepository {
-    return new MongoDeviceRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()), new MongoEntityConverter(LoggerFacade.createLogger()))
+  public static createCoreDeviceRepository(): CoreDeviceRepository {
+    return new MongoCoreDeviceRepository(MongoDatabase.getInstance(LoggerFacade.createLogger()), new MongoEntityConverter(LoggerFacade.createLogger()))
```

Comment:
The name of this method and the class you are instantiating are correct. The type of the method should still be `DeviceRepository`, not `CoreDeviceRepository`.